component/kyverno.jsonnet

```diff
@@ -37,85 +25,79 @@ local nodeSelectorConfig(role) =
       } ],
     };
 
-local patchRoleBindings = function(bindings)
-  std.map(
-    function(role_binding)
-      role_binding {
-        subjects: std.map(
-          function(subj) subj {
-            namespace: params.namespace,
-          },
-          super.subjects
-        ),
-      },
-    bindings
-  );
+local isOpenshift = std.startsWith(inv.parameters.facts.distribution, 'openshift');
+
+local monitoringLabel =
+  if isOpenshift then
+    {
+      'openshift.io/cluster-monitoring': 'true',
+    }
+  else
+    {
+      SYNMonitoring: 'main',
+    };
+
+local nodeSelectionNamespaceAnnotations = if params.nodeSelectorRole != null then
+  {
+    'openshift.io/node-selector': 'node-role.kubernetes.io/%s=' % params.nodeSelectorRole,
+  }
+else
+  {};
 
-local objects =
-  [] +
-  cluster_roles +
-  patchRoleBindings(cluster_role_bindings) +
-  roles +
-  patchRoleBindings(role_bindings) +
-  services +
-  [
-    service_account {},
-    deployment {
-      spec+: {
-        replicas: params.replicas,
-        template+: {
-          spec+: nodeSelectorConfig(params.nodeSelectorRole) + {
+com.Kustomization(
+  params.kustomization_url,
+  params.manifest_version,
+  {},
+  {
+    namespace: params.namespace,
+    replicas: [
+      {
+        name: 'kyverno',
+        count: params.replicas,
+      },
+    ],
+    patchesStrategicMerge: [
+      'deployment.yaml',
+      'namespace.yaml',
+    ],
+  },
+) {
+  // We need to patch namespace "kyverno" as the namespace renaming is only applied after doing strategic merge patches
```

Comment:
Moving this comment to before the `namespace:` key would make it easier to read and cross-reference.

Reply:
Ah yes you're right, that was moved to useless position :sweat_smile: